backend/src/register/register.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Body, Controller, Get, Param, Post } from '@nestjs/common';
+import { RegisterDto } from '../dto/register.dto';
+import { RegisterService } from './register.service';
+
+@Controller()
+export class RegisterController {
+    constructor(private readonly registerService: RegisterService) {}
+
+    @Post('register')
+    async registerBadge(@Body() { did, signature, service }: RegisterDto) {
+        const network = did.match(/^did:ethr:(.+):/)?.[1];
+
+        const txHash = await this.registerService.addService(
+            did,
+            network,
+            {
+                type: 'test',
+                serviceEndpoint: 'https://ipfs.io/ipfs/ID',
```

Comment:
The serviceEndpoint should be dynamic. Use the .env variable of IPFS server and ID will be the CID from IPFS add.